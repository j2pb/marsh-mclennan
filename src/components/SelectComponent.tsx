import React from "react";

interface SelectProps {
  options: string[];
  value: string;
  onChange: (value: string) => void;
}

const SelectComponent: React.FC<SelectProps> = ({ options, value, onChange }) => {
  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    onChange(event.target.value);
  };

  return (
    <select value={value} onChange={handleChange} >
      {options.map((option) => (
        <option key={option} value={option}>
          {option}
        </option>
      ))}
    </select>
  );
};

export default SelectComponent;
