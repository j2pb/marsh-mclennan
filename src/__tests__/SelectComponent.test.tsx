import { render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';
import SelectComponent from "../components/SelectComponent";

describe("SelectComponent", () => {
  const options = ["Washington", "California", "Florida"];
  const onChange = jest.fn();

  afterEach(() => {
    onChange.mockClear();
  });
//   it('should render correctly', () => {
//     const options = ["Washington", "California", "Florida"];
//     const value = 'California';
//     const onChange = jest.fn();

//     const wrapper = shallow(
//       <SelectComponent
//         options={options}
//         value={value}
//         onChange={onChange}
//       />
//     );

//     expect(wrapper).toMatchSnapshot();
//   });
  it('should render correctly', () => {
    const options = ["Washington", "California", "Florida"];
    const value = 'California';
    const onChange = jest.fn();
    const tree = renderer.create( <SelectComponent
               options={options}
               value={value}
               onChange={onChange}
             />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders with options", () => {
    const { getByDisplayValue, getByText } = render(
      <SelectComponent options={options} value={options[0]} onChange={onChange} />
    );

    expect(getByDisplayValue(options[0])).toBeInTheDocument();

    options.forEach((option) => {
      expect(getByText(option)).toBeInTheDocument();
    });
  });

  it("calls onChange when an option is selected", () => {
    const { getByDisplayValue, getByText } = render(
      <SelectComponent options={options} value={options[0]} onChange={onChange} />
    );

    const select = getByDisplayValue(options[0]);
    fireEvent.change(select, { target: { value: options[1] } });

    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(options[1]);
  });
});
