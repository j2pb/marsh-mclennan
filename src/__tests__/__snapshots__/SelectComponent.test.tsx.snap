// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectComponent should render correctly 1`] = `
<select
  onChange={[Function]}
  value="California"
>
  <option
    value="Washington"
  >
    Washington
  </option>
  <option
    value="California"
  >
    California
  </option>
  <option
    value="Florida"
  >
    Florida
  </option>
</select>
`;
